airbyte-integrations/connectors/source-notion/source_notion/components.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from dataclasses import dataclass, field, InitVar
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+import pendulum
+
+from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.declarative.extractors.record_filter import RecordFilter
+from airbyte_cdk.sources.declarative.retrievers import SimpleRetriever
+from airbyte_cdk.models import SyncMode
+
+
+@dataclass
+class NotionUserTransformation(RecordTransformation):
+    """
+    # TODO: Flesh out docstring
+    Custom transformation that conditionally moves the data in owner.{owner_type} 
+    to a new owner.info field when the record contains data for a "bot" type user.
+    """
+
+    def transform(self, record: MutableMapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
```

Comment:
does this need a custom transformation or could this be done with a combination of AddFields and RemoveFields?

Reply:
You're right! I had originally been struggling to get it working due to the conditional nature of the fields to be removed/added, but I just took another shot in the builder and seem to have it working, glad you pointed this out!